pkg/detector/detector.go

```diff
@@ -286,6 +281,12 @@ func (d *ResourceDetector) EventFilter(obj interface{}) bool {
 		return false
 	}
 
+	if clusterWideKey.Group == policyv1alpha1.GroupName {
+		if clusterWideKey.Kind == "PropagationPolicy" || clusterWideKey.Kind == "ClusterPropagationPolicy" {
+			return true
+		}
+	}
```

Comment:
> There will be a little bit complex when dealing with the EventFilter.

This is what I am worried about. It seems like a `glue`, I believe it can solve the problem, but the code becomes hard to understand.

Let's think about how to deal with the `event handler` thing better.

Reply:
I see, how about [moving the event filter to `Reconcile`](https://github.com/karmada-io/karmada/pull/1199/commits/19694c3db4fd9b198615492d6b3c79458298ba6e#diff-fdb6430a46e82ace9f449b971619671eac711f7a82d6b862c130236a65cd5101R203)?